spec/velveteen/worker_spec.rb

```diff
@@ -64,6 +64,61 @@
       _, properties, _ = queue.pop
       expect(properties[:headers]).to eq("foo" => "bar")
     end
+
+    describe "payload validation" do
+      it "raises an error when payload does not match schema" do
+        message = Velveteen::Message.new(
+          data: {"foo" => "bar"},
+          headers: {}
+        )
+        invalid_message = Velveteen::Message.new(
```

Comment:
What do you think about not building a message object for this and renaming it to `invalid_payload`?

Reply:
ha I was thinking that as I wrote this but passed over it thank you!